block/submit.go

```diff
@@ -35,14 +36,18 @@ func (m *Manager) SubmitLoop(ctx context.Context,
 // SubmitLoopInner is a unit testable impl of SubmitLoop
 func SubmitLoopInner(ctx context.Context,
 	bytesProduced chan int, // a channel of block and commit bytes produced
-	maxBatchSkew uint64, // max number of batches that submitter is allowed to have pending
+	maxBlockSkew uint64, // max number of blocks that submitter is allowed to have pending
+	pendingSubmittedBlocks func() uint64,
 	maxBatchTime time.Duration, // max time to allow between batches
 	maxBatchBytes uint64, // max size of serialised batch in bytes
 	createAndSubmitBatch func(maxSizeBytes uint64) (sizeBlocksCommits uint64, err error),
 ) error {
 	eg, ctx := errgroup.WithContext(ctx)
 
 	pendingBytes := atomic.Uint64{}
+
+	pendingBlocks := atomic.Uint64{}
```

Comment:
not sure I understand why we need to define an atomic value for it for saving, loading and  updating it vs just call `GetUnsubmittedBlocks` when necessary (i.e to just to check skew and possibly metrics)?